### ISSUE TITLE
[Bug]: Nixpacks static site does not respect 200.html

### ISSUE THREAD

**Issue by AlejandroAkbal** [coollabsio/coolify#3039]
### Description

I'm having the same issue as #2562
![image](https://github.com/user-attachments/assets/9b4d6de7-87a4-4b5f-8415-8ddd50bc009d)

Seems like Nuxt 3 with `generate` does not work
You can access the files directly, but index.html is the nginx default page, 200.html is not loaded
![image](https://github.com/user-attachments/assets/5a319aac-f9eb-4878-9879-1cefe77108d5)

These are my settings
![image](https://github.com/user-attachments/assets/ae338e06-2116-4735-816a-2e0d4294a2e3)

### Minimal Reproduction (if possible, example repository)

You can use this repo
https://github.com/Rule-34/App [NSFW]

### Exception or Error

Nginx default page 

![image](https://github.com/user-attachments/assets/9b4d6de7-87a4-4b5f-8415-8ddd50bc009d)

### Version

 v4.0.0-beta.323

### Cloud?

- [ ] Yes
- [X] No

**Comment by Thijmen** [comment]
By default, nginx will load index.html. It seems that your app does not generate an index.html?

**Comment by AlejandroAkbal** [comment]
> By default, nginx will load index.html. It seems that your app does not generate an index.html?

No, Nuxt does not generate index, it generates a 200.html and 404.html

**Comment by AlejandroAkbal** [comment]
Yeah, that would work, but isn’t there an option to support these modern frameworks that generate 200.html, 400.html and 500.html? As seen on my initial screenshot

**Comment by justserdar** [comment]
It does build a index.html file along with the 200.html, 50x and 404.html page.

![image](https://github.com/user-attachments/assets/dd09ae92-1cfd-45cf-8492-910c01c718c4)
Try the above settings

**Comment by AlejandroAkbal** [comment]
`/nuxt/static` path does not work, my settings are the same as yours
![image](https://github.com/user-attachments/assets/e3da7d67-bd90-498e-aa13-dcaa8bc80657)

Please try to deploy my repo https://github.com/Rule-34/App and you'll see default nginx in index.html

**Comment by AlejandroAkbal** [comment]
One thing I noticed is I dont have watch paths on my UI?
![image](https://github.com/user-attachments/assets/7c9b6b85-5db2-4a38-b56a-1cdb49a1cf91)

**Comment by tkbstudios** [comment]
Disgusting.

**Comment by Giveback007** [comment]
This is also an issue for me. I'm using a SvelteKit app with adapter-static and encountering routing problems.

My setup:
- SvelteKit app with adapter-static
- Fallback: 200.html (can't use index.html as it overrides pre-rendered content)
- Partially pre-rendered, the rest is CSR
- Need "Is it static site? = true"

Problem:
1. Pre-rendered index.html doesn't route files correctly
2. URLs like `my.domain.com/_app/path-to-file.js` become `my.domain.com/admin/_app/path-to-file.js` (depending on parent path)
3. It loads index.html as the content for these files, treating it as a fallback

Attempted solution:
I tried a post-build script to change URLs in index.html:

```js
const scriptDir = dirname(fileURLToPath(import.meta.url));
const indexPath = join(scriptDir, '../build/index.html');

const indexContents = readFileSync(indexPath, 'utf8');
writeFileSync(indexPath, indexContents.replaceAll('./_app/', '/_app/'));
```

However, this caused all URLs to load index.html.
I'm unsure how to solve this correctly. It's also odd that it defaults to index.html, as any unavailable URLs (in a completely pre-rendered site) would load index.html, whereas a 404.html might be preferable for some cases.
Any suggestions on how to address this routing issue would be appreciated.

**Comment by Giveback007** [comment]
This guide resolved my problem:
https://billyle.dev/posts/fix-missing-404-pages-for-coolify-static-site-deployments

Essentially you'll have to dockerize your app:
``` Dockerfile
ARG NODE_VERSION=22.8.0
ARG PNPM_VERSION=9.10.0

FROM node:${NODE_VERSION}-alpine AS builder

WORKDIR /app

COPY package.json pnpm-lock.yaml* ./
RUN npm install -g pnpm@${PNPM_VERSION}
RUN pnpm install

COPY . .
RUN pnpm build

FROM nginx:stable-alpine3.17 AS final
COPY nginx.conf /etc/nginx/nginx.conf
COPY --from=builder /app/build /usr/share/nginx/html
EXPOSE 4321
```

``` nginx.conf
worker_processes  1;

events {
  worker_connections  1024;
}

http {
  server {
    listen 4321;
    server_name   _;

    root   /usr/share/nginx/html;
    index  index.html index.htm;
    include /etc/nginx/mime.types;

    gzip on;
    gzip_min_length 1000;
    gzip_proxied expired no-cache no-store private auth;
    gzip_types text/plain text/css application/json application/javascript application/x-javascript text/xml application/xml application/xml+rss text/javascript;

    error_page 404 /404.html;
    location = /404.html {
            root /usr/share/nginx/html;
            internal;
    }

    location / {
            try_files $uri $uri/index.html /CSR.html;
    }
  }
}
```

In coolify set:
Network / Ports Exposes
4321

(Modify the prams to fit your app)
---

In the above example CSR.html is my fallback html, but it can be anything (for example 200.html), this config also handles 404.html, but in my case it should always fallback to CSR.html (in which it will handle the 404)

---

Hope this solve a big headache for anyone else encountering this issue.

**Comment by Keromudo** [comment]
## 💬 Error description
I've encountered the same problem with my Nuxt3 app. I've used the setup from [this](https://github.com/coollabsio/coolify-examples/tree/main/nuxt/static) repo and did everything by the [docs](https://coolify.io/docs/applications/nuxt).
![Image](https://github.com/user-attachments/assets/f61d85de-1ef6-4941-b817-6a12af13ce00)
Problem seems to be in nixpacks. It will use _npm run build_ as build command regardless of static site checked or not. Thus .html files are not generated.

![Image](https://github.com/user-attachments/assets/83cd4732-e221-41a4-ae0a-dd6e2aed0a04)
## ⭐ Solution
You just need to configure the build command to use _npm run generate_ and everything works as expected. This config works for me.
 
![Image](https://github.com/user-attachments/assets/050b0247-9ec4-42bf-9861-ce315bb8c25a)
and voila!

![Image](https://github.com/user-attachments/assets/2b4991ec-0ba2-4c6f-8d9d-85d6019f4811)

> [!TIP]
> You can setup health check like this and check if _.output/public_ was generated or not.
>
> ![Image](https://github.com/user-attachments/assets/6224b24e-266b-428e-a98c-8120bacb6133)

> [!IMPORTANT]
> I am using v4.0.0-beta.360

**Comment by Cinzya** [comment]
@AlejandroAkbal is this still a problem or can we close?
Coolify now allows you to edit the Nginx Configuration when you have `Is it a static site?` enabled. That should in theory solve your problem.

![Image](https://github.com/user-attachments/assets/43263f2a-ce12-4ee6-833e-04a4a2f18ddc)

**Comment by AlejandroAkbal** [comment]
Could be, I abandoned having a static website since this didn't work, but if you can modify it, then yeah

Feel free to close
